pkg/cluster/delete.go

```diff
@@ -425,13 +426,12 @@ func (m *manager) deleteFederatedCredentials(ctx context.Context) error {
 				&armmsi.FederatedIdentityCredentialsClientDeleteOptions{},
 			)
 			if err != nil {
-				cloudErr := err.(*api.CloudError)
-
-				if cloudErr.StatusCode == http.StatusNotFound {
-					m.log.Errorf("federated identity credentials not found for %s: %v", identity.ResourceID, cloudErr.Error())
+				cloudErr, ok := err.(*azcore.ResponseError)
```

Comment:
Are you able to use `errors.As()`? You don't have to worry about a panic and can handle wrapped errors as well.

Reply:
I'll do this, thanks for the suggestion!